nose2/plugins/mp.py

```diff
@@ -274,6 +274,12 @@ def _localize(self, event):
                     'test_not_found',
                     RuntimeError("Unable to locate test case for %s in "
                                  "main process" % event.test))._tests[0]
+            # subtest support
+            if 'subtest' in event.metadata:
+                message, params = event.metadata.pop('subtest')
+                if type(message) == type(object()):
+                    message = unittest.case._subtest_msg_sentinel
```

Comment:
Yes, this is exactly it. The sentinel does not survive the pickle round-trip and has to be reconstructed.

I have tried several variants before settling on this. At one point I used an `Event.subtest` attribute, another time I tried to be clever with `__setstate__`... 

But if you think about it, `mp._localize` effectively _is_ the Event's `__setstate__`. Loader, runner, and result are reconstructed there, even the test object itself is "unpickled" in `_localize`. And the `_SubTest` wrapper is applied there as well of course.

This is why I decided to keep `Event` modifications to a minimum and have all the unpickling code in one place instead of spreading it over multiple files. Also, you did understand the intent perfectly which makes me confident others will too. 😃

Reply:
Hm. I'm not sure I _like_ the fact that the MP plugin is responsible for these operations, but you're right that it is today... I'm going to think more about this, but it doesn't need any change today.

I think we should add a comment to this section for future readers.
Although I was able to follow, I wasn't certain that I was correct until you confirmed it.

Reply:
I have added a comment. Thank you for your feedback.